packages/tradershub/src/hooks/useMT5AccountHandler.ts

```diff
@@ -16,35 +16,40 @@ const useMT5AccountHandler = () => {
         error: isCreateMT5AccountError,
         isLoading: createMT5AccountLoading,
         isSuccess: isCreateMT5AccountSuccess,
+        status: createMT5AccountStatus,
         mutate: createMT5Account,
     } = useCreateMT5Account();
     const { isLoading: tradingPlatformPasswordChangeLoading, mutateAsync: tradingPasswordChange } =
         useTradingPlatformPasswordChange();
     const { data: activeTrading } = useActiveTradingAccount();
     const { data: settings } = useSettings();
     const { data: availableMT5Accounts } = useAvailableMT5Accounts();
-    const isMT5PasswordNotSet = accountStatus?.is_mt5_password_not_set;
-
     const { cfdState } = useCFDContext();
+
     const { marketType: marketTypeState, selectedJurisdiction } = cfdState;
     const marketType = marketTypeState ?? MarketType.ALL;
+    const isMT5PasswordNotSet = accountStatus?.is_mt5_password_not_set;
 
     const accountType = marketType === MarketType.SYNTHETIC ? 'gaming' : marketType;
     const categoryAccountType = activeTrading?.is_virtual ? Category.DEMO : accountType;
-    const handleSubmit = (password: string) => {
-        // in order to create account, we need to set a password through trading_platform_password_change endpoint first
-        // then only mt5_create_account can be called, otherwise it will response an error for password required
+
+    const doesNotMeetPasswordPolicy =
+        isCreateMT5AccountError?.error?.code === 'InvalidTradingPlatformPasswordFormat' ||
+        isCreateMT5AccountError?.error?.code === 'IncorrectMT5PasswordFormat';
+
+    // in order to create account, we need to set a password through trading_platform_password_change endpoint first
+    // then only mt5_create_account can be called, otherwise it will response an error for password required
+    const handleSubmit = async (password: string) => {
         if (isMT5PasswordNotSet) {
-            return tradingPasswordChange({
+            await tradingPasswordChange({
                 new_password: password,
                 platform: CFDPlatforms.MT5,
-            }).then(() => {
-                return createPassword(password);
             });
         }
 
-        createPassword(password);
+        await createPassword(password);
```

Comment:
You are awaiting here but the createPassword function below does not have async. There is a lot of problems with the way you guys are using useMutation? If you want to asynchronously mutate, you should useMutateAsync for `useCreateMT5Account` so that the mutate function (`createMT5Account`) is asynchronous